components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/DocumentWrapperSupport.java

```diff
@@ -16,35 +16,36 @@
 
 import javax.annotation.Nullable;
 
+import com.google.common.collect.ImmutableSet;
 import com.orientechnologies.orient.core.id.ORID;
-import com.tinkerpop.blueprints.impls.orient.OrientVertex;
+import com.orientechnologies.orient.core.record.impl.ODocument;
 import org.joda.time.DateTime;
 
 import static com.google.common.base.Preconditions.checkNotNull;
 import static com.google.common.base.Preconditions.checkState;
 
 /**
- * Supporting base class for {@link VertexWrapper} implementations.
+ * Supporting base class for {@link DocumentWrapper} implementations.
  *
  * @since 3.0
  */
-abstract class VertexWrapperSupport
-    implements VertexWrapper
+abstract class DocumentWrapperSupport
+    implements DocumentWrapper
```

Comment:
They are not similar. The Entity(Support) keeps only refs to rid/version and can be used outside transaction, while this keeps the ODocument itself and can be used only within the transaction.

Extraction of EntityType outside of EntityAdapter was in order to reuse the type registrations bits. The rest from EntityAdapter (read/write from entity to Document) were not applicable.

Going for using EntityAdapter and read/write ODocument <-> Entity will require much more changes with no real benefit.